### PR TITLE
New system status security section

### PR DESCRIPTION
**Pull request by claudiosanches**
Ref #12279

### CODE REVIEW

File: includes/admin/views/html-admin-page-status-report.php [hunk 1]
```diff
@@ -330,6 +330,37 @@
 <table class="wc_status_table widefat" cellspacing="0">
 	<thead>
 		<tr>
+			<th colspan="3" data-export-label="Security"><h2><?php _e( 'Security', 'woocommerce' ); ?></h2></th>
+		</tr>
+	</thead>
+	<tbody>
+		<tr>
+			<td data-export-label="Under HTTPS"><?php _e( 'Under HTTPS', 'woocommerce' ); ?>:</td>
+			<td class="help"><?php echo wc_help_tip( __( 'Are your shop under HTTPS?', 'woocommerce' ) ); ?></td>
```
Comment (mikejolley) [inline includes/admin/views/html-admin-page-status-report.php]: Is the connection to your store secure?

File: includes/admin/views/html-admin-page-status-report.php [hunk 2]
```diff
@@ -330,6 +330,37 @@
 <table class="wc_status_table widefat" cellspacing="0">
 	<thead>
 		<tr>
+			<th colspan="3" data-export-label="Security"><h2><?php _e( 'Security', 'woocommerce' ); ?></h2></th>
+		</tr>
+	</thead>
+	<tbody>
+		<tr>
+			<td data-export-label="Under HTTPS"><?php _e( 'Under HTTPS', 'woocommerce' ); ?>:</td>
```
Comment (mikejolley) [inline includes/admin/views/html-admin-page-status-report.php]: Secure connection (HTTPS)

File: includes/admin/views/html-admin-page-status-report.php [hunk 3]
```diff
@@ -330,6 +330,37 @@
...
+					<mark class="yes"><span class="dashicons dashicons-yes"></span></mark>
+				<?php endif; ?>
+			</td>
+		</tr>
+		<tr>
+			<td data-export-label="Display errors disabled"><?php _e( 'Display errors disabled', 'woocommerce' ); ?></td>
+			<td class="help"><?php echo wc_help_tip( __( 'Error messages frequently contain sensitive information about your shop environment, and should never be presented to untrusted sources.', 'woocommerce' ) ); ?></td>
+			<td>
+				<?php if ( ( defined( 'WP_DEBUG' ) && defined( 'WP_DEBUG_DISPLAY' ) && WP_DEBUG && WP_DEBUG_DISPLAY ) || 0 !== intval( ini_get( 'display_errors' ) ) ) : ?>
+					<mark class="error"><span class="dashicons dashicons-warning"></span><?php _e( 'All error messages should never be displayed if not in a closed development environment.', 'woocommerce' ); ?></mark>
```
Comment (mikejolley) [inline includes/admin/views/html-admin-page-status-report.php]: Error messages should not be shown to visitors if this is a live store.

File: includes/admin/views/html-admin-page-status-report.php [hunk 4]
```diff
@@ -330,6 +330,37 @@
...
+			<td>
+				<?php if ( $settings['force_ssl'] || 'https' !== substr( get_permalink( wc_get_page_id( 'shop' ) ), 0, 5 ) ) : ?>
+					<mark class="error"><span class="dashicons dashicons-warning"></span><?php printf( __( 'Your shop is not under HTTPS. <a href="%s" target="_blank">Learn more about HTTPS and SSL Certificates</a>.', 'woocommerce' ), 'https://docs.woocommerce.com/document/ssl-and-https/' ); ?></mark>
+				<?php else : ?>
+					<mark class="yes"><span class="dashicons dashicons-yes"></span></mark>
+				<?php endif; ?>
+			</td>
+		</tr>
+		<tr>
+			<td data-export-label="Display errors disabled"><?php _e( 'Display errors disabled', 'woocommerce' ); ?></td>
```
Comment (mikejolley) [inline includes/admin/views/html-admin-page-status-report.php]: Hide errors from visitors

File: includes/admin/views/html-admin-page-status-report.php [hunk 5]
```diff
@@ -330,6 +330,37 @@
...
+			<td class="help"><?php echo wc_help_tip( __( 'Error messages frequently contain sensitive information about your shop environment, and should never be presented to untrusted sources.', 'woocommerce' ) ); ?></td>
```
Comment (mikejolley) [inline includes/admin/views/html-admin-page-status-report.php]: Error messages can contain sensitive information about your store environment. These should be hidden from untrusted visitors.

File: includes/admin/views/html-admin-page-status-report.php [hunk 6]
```diff
@@ -330,6 +330,37 @@
...
+			<th colspan="3" data-export-label="Security"><h2><?php _e( 'Security', 'woocommerce' ); ?></h2></th>
+		</tr>
+	</thead>
+	<tbody>
+		<tr>
+			<td data-export-label="Under HTTPS"><?php _e( 'Under HTTPS', 'woocommerce' ); ?>:</td>
+			<td class="help"><?php echo wc_help_tip( __( 'Are your shop under HTTPS?', 'woocommerce' ) ); ?></td>
+			<td>
+				<?php if ( $settings['force_ssl'] || 'https' !== substr( get_permalink( wc_get_page_id( 'shop' ) ), 0, 5 ) ) : ?>
+					<mark class="error"><span class="dashicons dashicons-warning"></span><?php printf( __( 'Your shop is not under HTTPS. <a href="%s" target="_blank">Learn more about HTTPS and SSL Certificates</a>.', 'woocommerce' ), 'https://docs.woocommerce.com/document/ssl-and-https/' ); ?></mark>
```
Comment (mikejolley) [inline includes/admin/views/html-admin-page-status-report.php]: Your store is not using HTTPS.

File: includes/admin/views/html-admin-page-status-report.php [hunk 7]
```diff
@@ -330,6 +330,37 @@
...
 		<tr>
+			<th colspan="3" data-export-label="Security"><h2><?php _e( 'Security', 'woocommerce' ); ?></h2></th>
+		</tr>
+	</thead>
+	<tbody>
+		<tr>
+			<td data-export-label="Under HTTPS"><?php _e( 'Under HTTPS', 'woocommerce' ); ?>:</td>
+			<td class="help"><?php echo wc_help_tip( __( 'Are your shop under HTTPS?', 'woocommerce' ) ); ?></td>
+			<td>
+				<?php if ( $settings['force_ssl'] || 'https' !== substr( get_permalink( wc_get_page_id( 'shop' ) ), 0, 5 ) ) : ?>
```
Comment (mikejolley) [inline includes/admin/views/html-admin-page-status-report.php]: First conditonal looks wrong.

**Review by mikejolley — DISMISSED** [review]
Added some feedback.

**Comment by justinshreve** [comment]
This info should also be present in the API. See `WC_REST_System_Status_Controller`.

**Comment by claudiosanches** [comment]
@justinshreve Oopps, complete forget about the API.

**Comment by claudiosanches** [comment]
Included REST API support and tests.

**Pull request merged**